File: app/src/main/java/com/android/ash/charactersheet/dac/dao/android/AndroidPreferenceDao.java
package com.android.ash.charactersheet.dac.dao.android;

import android.content.Context;
import android.content.SharedPreferences;

import com.android.ash.charactersheet.dac.dao.PreferenceDao;

/**
 * Access object to preferences using the Android Preferences API.
 */
public class AndroidPreferenceDao implements PreferenceDao {

    /** The name of the file containing the preferences. */
    public static final String FILENAME = "d20cs.prefs";

    private final SharedPreferences preferences;
    private final SharedPreferences.Editor editor;

    /**
     * Instantiates AndroidPreferenceDao.
     * 
     * @param context
     *            The application context.
     */
    public AndroidPreferenceDao(final Context context) {
        preferences = context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE);
        editor = preferences.edit();
    }

    @Override
    public boolean getBoolean(final String key, final boolean defaultValue) {
        return preferences.getBoolean(key, defaultValue);
    }

    @Override
    public int getInt(final String key, final int defaultValue) {
        return preferences.getInt(key, defaultValue);
    }

    @Override
    public void setBoolean(final String key, final boolean value) {
        editor.putBoolean(key, value);
        editor.commit();
    }

    @Override
    public void setInt(final String key, final int value) {
        editor.putInt(key, value);
        editor.commit();
    }

}

File: app/src/main/java/com/android/ash/charactersheet/dac/dao/android/AndroidPreferenceDao.kt
package com.android.ash.charactersheet.dac.dao.android

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import com.android.ash.charactersheet.dac.dao.PreferenceDao

/**
 * Access object to preferences using the Android Preferences API.
 *
 * @param context The application context.
 */

class AndroidPreferenceDao @JvmOverloads constructor(context: Context, filename: String = FILENAME) : PreferenceDao {

    private val preferences: SharedPreferences = context.getSharedPreferences(filename, Context.MODE_PRIVATE)
    private val editor: Editor = preferences.edit()

    override fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return preferences.getBoolean(key, defaultValue)
    }

    override fun getInt(key: String, defaultValue: Int): Int {
        return preferences.getInt(key, defaultValue)
    }

    override fun setBoolean(key: String, value: Boolean) {
        editor.putBoolean(key, value)
        editor.commit()
    }

    override fun setInt(key: String, value: Int) {
        editor.putInt(key, value)
        editor.commit()
    }

    companion object {
        /** The name of the file containing the preferences.  */
        const val FILENAME = "d20cs.prefs"
    }

}